src/test/java/seedu/address/storage/JsonAdaptedLeaveTest.java

```diff
@@ -1,111 +1,145 @@
+    }
+
+    @Test
+    public void toModelType_emptyTitle_throwsIllegalValueException() {
+        JsonAdaptedLeave leave = new JsonAdaptedLeave(VALID_START, VALID_END, EMPTY_TITLE, VALID_DESCRIPTION,
+                VALID_STATUS, VALID_EMPLOYEE);
+        assertThrows(IllegalValueException.class, Title.MESSAGE_CONSTRAINTS, leave::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidTitle_throwsIllegalValueException() {
+        JsonAdaptedLeave leave = new JsonAdaptedLeave(VALID_START, VALID_END, INVALID_TITLE, VALID_DESCRIPTION,
+                VALID_STATUS, VALID_EMPLOYEE);
+        assertThrows(IllegalValueException.class, Title.MESSAGE_CONSTRAINTS, leave::toModelType);
+    }
+
+    @Test
+    public void toModelType_nullDescription_throwsIllegalValueException() {
+        JsonAdaptedLeave leave = new JsonAdaptedLeave(VALID_START, VALID_END, VALID_TITLE, null,
+                VALID_STATUS, VALID_EMPLOYEE);
+        String expectedMessage = String.format(JsonAdaptedLeave.MISSING_FIELD_MESSAGE_FORMAT, "description");
+        assertThrows(IllegalValueException.class, expectedMessage, leave::toModelType);
+    }
+
+    @Test
+    public void toModelType_emptyDescription_success() throws Exception {
+        JsonAdaptedLeave leave = new JsonAdaptedLeave(VALID_START, VALID_END, VALID_TITLE, EMPTY_DESCRIPTION,
+                VALID_STATUS, VALID_EMPLOYEE);
+        assertTrue(VALID_LEAVE.isSameLeave(leave.toModelType()));
+    }
+
+    @Test
+    public void toModelType_invalidDescription_throwsIllegalValueException() {
+        JsonAdaptedLeave leave = new JsonAdaptedLeave(VALID_START, VALID_END, VALID_TITLE, INVALID_DESCRIPTION,
+                VALID_STATUS, VALID_EMPLOYEE);
+        assertThrows(IllegalValueException.class, Description.MESSAGE_CONSTRAINTS, leave::toModelType);
+    }
+
+    @Test
+    public void toModelType_nullEmployee_throwsIllegalValueException() {
+        assertThrows(NullPointerException.class, () -> new JsonAdaptedLeave(VALID_START, VALID_END, VALID_TITLE,
+                VALID_DESCRIPTION, VALID_STATUS, null));
+    }
+
+    @Test
+    public void toModelType_nullStatus_throwsIllegalValueException() {
+        JsonAdaptedLeave leave = new JsonAdaptedLeave(VALID_START, VALID_END, VALID_TITLE, VALID_DESCRIPTION,
+                null, VALID_EMPLOYEE);
+        String expectedMessage = String.format(JsonAdaptedLeave.MISSING_FIELD_MESSAGE_FORMAT, "status");
+        assertThrows(IllegalValueException.class, expectedMessage, leave::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidStatus_throwsIllegalValueException() {
+        JsonAdaptedLeave leave = new JsonAdaptedLeave(VALID_START, VALID_END, VALID_TITLE, VALID_DESCRIPTION,
+                INVALID_STATUS, VALID_EMPLOYEE);
+        String expectedMessage = Status.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalArgumentException.class, expectedMessage, leave::toModelType);
+    }
```

Comment:
Maybe we can also test for empty start/end date, empty employee, invalid employee, and empty status?